app/scripts/models/reliers/oauth.js

```diff
@@ -16,9 +16,49 @@ define(function (require, exports, module) {
   var RelierKeys = require('lib/relier-keys');
   var Url = require('lib/url');
   var Validate = require('lib/validate');
+  var Validator = require('vat');
 
   var RELIER_FIELDS_IN_RESUME_TOKEN = ['state', 'verificationRedirect'];
 
+  /*eslint-disable camelcase*/
+  var CLIENT_INFO_SCHEMA = {
+    id: Validator.string().required().use(Validate.isHexValid).as('clientId'),
+    image_uri: Validator.string().optional().allow('').use(Validate.isUrlValid).as('imageUri'),
+    name: Validator.string().required().min(1).as('serviceName'),
+    privacy_uri: Validator.string().optional().allow('').use(Validate.isUriValid).as('privacyUri'),
+    redirect_uri: Validator.string().required().use(Validate.isUriValid).as('redirectUri'),
+    terms_uri: Validator.string().optional().allow('').use(Validate.isUriValid).as('termsUri'),
+    trusted: Validator.boolean().required()
+  };
+
+  var SIGNIN_SIGNUP_QUERY_PARAM_SCHEMA = {
+    access_type: Validator.string().optional().use(Validate.isAccessTypeValid).as('accessType'),
+    client_id: Validator.string().required().use(Validate.isHexValid).as('clientId'),
+    keys: Validator.boolean().optional(),
+    prompt: Validator.string().optional().use(Validate.isPromptValid),
+    redirectTo: Validator.string().optional().use(Validate.isUriValid),
+    redirect_uri: Validator.string().optional().use(Validate.isUriValid).as('redirectUri'),
+    scope: Validator.string().required().min(1),
+    service: Validator.string().optional(),
+    state: Validator.string().optional(),
+    verification_redirect: Validator.string().optional().use(Validate.isVerificationRedirectValid).as('verificationRedirect')
+  };
+
+  var VERIFICATION_INFO_SCHEMA = {
+    access_type: Validator.string().optional().use(Validate.isAccessTypeValid).as('accessType'),
```

Comment:
I'd like to add a way to define generic types that are reusable, something like:

```js
Validator.register('access_type', Validator.string().use(Validate.isAcessTypeValid))

...
var VERIFICATION_INFO_SCHEMA = {
  access_type: Validator.access_type().optional().as('accessType'),
...
```